### PR TITLE
Update javadoc for Subscribers.empty()

### PR DESCRIPTION
**Pull request by digitalbuddha**
fix for https://github.com/ReactiveX/RxJava/issues/3664

### CODE REVIEW

File: src/main/java/rx/observers/Subscribers.java [hunk 1]
```diff
@@ -31,10 +31,9 @@ private Subscribers() {
     }
 
     /**
-     * Returns an inert {@link Subscriber} that does nothing in response to the emissions or notifications from
-     * any {@code Observable} it subscribes to. This is different, however, from an {@code EmptyObserver}, in
-     * that it will throw an exception if its {@link Subscriber#onError onError} method is called (whereas
-     * {@code EmptyObserver} will swallow the error in such a case).
+     * Returns an inert {@link Subscriber} that does nothing in response to the emissions or notifications 
+     * from any {@code Observable} it subscribes to.  Will throw an exception if {@link Subscriber#onError onError} 
+     * method is called
```
Comment (zsxwing) [inline src/main/java/rx/observers/Subscribers.java]: Could also add the exception name in the doc, like Will throw an **{@link OnErrorNotImplementedException}** ...

**Comment by artem-zinnatullin** [comment]
:+1:

**Comment by akarnokd** [comment]
:+1:

**Comment by digitalbuddha** [comment]
I'll squash in a few when I get to work.

**Comment by stevegury** [comment]
:+1:

**Comment by zsxwing** [comment]
:+1:

**Pull request merged**